test/src/e2e_vm_tests/test_programs/should_fail/ref_mutable_fn_args_struct/test.toml

```diff
@@ -0,0 +1,3 @@
+category = "fail"
+
+# check: $()This parameter was declared as a reference, which is implied for non-primitive types.
```

Comment:
While this is true, I think we still want the `ref` keyword for reference types. Basically, my understanding is that we want the `ref` keyword to be added anywhere we had `mut` only before. So, our options are:
```rust
fn mut_arg_non_primitive(ref mut s: S) {
    s = S{}
}
fn mut_arg_primitive(ref mut b: bool) {
    b = true;
}
fn mut_arg_non_primitive(s: S) {
    // Can't modify `s` here
}
fn mut_arg_primitive(b: bool) {
    // Can't modify `b` here
}
```

Reply:
Oh dear, so tricky.  Syntactically I thought the only changes we're making were mut args now _must_ have the `ref` qualifier, but this is _only_ for args, not local variables, and yeah, for all mut args regardless of their type.

So for mut args, it's always `ref mut` and for mut locals it's always just `mut`.